### ISSUE TITLE
[Bug] Maybe a finetune bug

### ISSUE THREAD

**Issue by HAOCHENYE** [facebookresearch/llama-recipes#27]
`lr_scheduler.step` is called every `gradient_accumulation_steps` times. This leads to a rapid decrease of the learning rate, which I believe is a bug.

https://github.com/facebookresearch/llama-recipes/blob/cd82118b74d2fd739bd6227af33b661d04a97406/utils/train_utils.py#L108

**Comment by HamidShojanazeri** [comment]
thanks for opening the issue, addressed in the PR #28.

**Comment by thtang** [comment]
![Screenshot 2023-07-22 at 6 17 38 PM](https://github.com/facebookresearch/llama-recipes/assets/15909789/7742e5e4-f879-487b-a866-538661bca65e)
Is it expected to have a bigger learning rate after the bug has been fixed?

**Comment by HamidShojanazeri** [comment]
@thtang is this from the notebook/ which setting your are running?

**Comment by thtang** [comment]
Hi @HamidShojanazeri I just ran the quickstart.ipynb without changing any settings. It seems that the learning rate will decrease from 1*e-4 (or 9.xx*e-5) to 5.xx*e-6 in one epoch.

**Comment by HamidShojanazeri** [comment]
@thtang I see, the quickstart notebook is using HF trainer so it would not use the training loop provided in utils/train_utils.py so basically the above fix would not touch this notebook.

Note that, the optimizer used in the notebook is `adamw_torch_fused` and the gamma being used is the default in HF trainer, so you can control that with setting optimizer hyper-parameters for HF trainer. I guess if you get the desired output its what matters, default setting in the notebook been set based on the results we observed. So to further control the lr is needed please check [HF trainer] (https://huggingface.co/docs/transformers/main_classes/trainer) optimizer settings.

**Comment by thtang** [comment]
Oh got it. Many thanks!
